packages/web-api/src/types/helpers.ts

```diff
@@ -0,0 +1,2 @@
+/** Omit all keys K from possible union types T */
+export type ExcludeFromUnion<T, K extends string> = T extends T ? Omit<T, K> : never;
```

Comment:
`Omit` seems to act strange with union types so instead this helper type is added! The `T extends T` is used to force the type checker to apply the `Omit` to each member of the union.